### ISSUE TITLE
Modal Text Inaccessible on iOS

### ISSUE THREAD

**Issue by mdparker26** [nico3333fr/jquery-accessible-modal-window-aria#21]
Hi -

I've recently noticed a problem with the modals on https://a11y.nicolas-hoffmann.net/modal/ where the text is not read on iOS 10.3.3 devices when using Safari and VoiceOver.  

For example, on the "First Simple Example" modal, the following is what appears: 

Close (This is read)
Title (This is read)
Hello world!  (This is not read)

Any idea what could be happening?

Thanks, 
Michael

**Comment by nico3333fr** [comment]
Hi @mdparker26 

thanks for this report, I asked @goetsu if he could reproduce it (I'm not using a mac), and he confirmed the bug. We are going to make some tests, to know if it is an iOS bug or if we can find a workaround to avoid it.

Thanks,
Nicolas

**Comment by nico3333fr** [comment]
It seems that the problem is coming from the use of ```aria-modal="true"``` with ```role="dialog" ```. 

While waiting for a perfect solution, I guess we will have to remove role="dialog", and separate cases.

**Comment by nico3333fr** [comment]
Could you test on this example ? https://a11y.nicolas-hoffmann.net/modal/

**Comment by nico3333fr** [comment]
It seems that the fix worked, could you confirm ?

**Comment by mdparker26** [comment]
Hi Nicolas – 

 

Thanks for your work to resolve this issue.  I just tested on iOS 10.3.3 with an iPhone 7, Safari and VoiceOver and everything is working fine.  I have not tested with other browsers yet but will later today and let you know what I find. 

 

Thanks again,

 

 

Michael Parker, MS

 <mailto:michaelp@accessingenuity.com> michaelp@accessingenuity.com

 

Access Ingenuity

4751 Hoen Avenue 

Santa Rosa, CA 95405

T: 877-579-4380

C: 707-508-6589

F: 707-579-4273

 

 <http://www.accessingenuity.com/> www.accessingenuity.com

 <http://www.documentaccessibility.com> www.documentaccessibility.com

**Comment by mdparker26** [comment]
Hi Nicolas – 

 

I just tested this with the following browser/screen reader combinations on Windows 10:  JAWS:  IE11/Firefox/Chrome; NVDA: Firefox.  The functionality is working fine.  The one issue I found is that the last “close it” is not identified as a button or link to screen reader users: 

 

<span id="js-modal-overlay" class="simple-modal-overlay" title="Close this window" data-background-click="enabled">

<span class="invisible">Close it</span>

</span>

 

Thanks,

 

Michael

 

From: Michael Parker [mailto:michaelp@accessingenuity.com] 
Sent: Thursday, August 17, 2017 8:57 AM
To: 'nico3333fr/jquery-accessible-modal-window-aria' <reply@reply.github.com>; 'nico3333fr/jquery-accessible-modal-window-aria' <jquery-accessible-modal-window-aria@noreply.github.com>
Cc: 'Mention' <mention@noreply.github.com>
Subject: RE: [nico3333fr/jquery-accessible-modal-window-aria] Modal Text Inaccessible on iOS (#21)

 

Hi Nicolas – 

 

Thanks for your work to resolve this issue.  I just tested on iOS 10.3.3 with an iPhone 7, Safari and VoiceOver and everything is working fine.  I have not tested with other browsers yet but will later today and let you know what I find. 

 

Thanks again,

 

 

Michael Parker, MS

 <mailto:michaelp@accessingenuity.com> michaelp@accessingenuity.com

 

Access Ingenuity

4751 Hoen Avenue 

Santa Rosa, CA 95405

T: 877-579-4380

C: 707-508-6589

F: 707-579-4273

 

 <http://www.accessingenuity.com/> www.accessingenuity.com

 <http://www.documentaccessibility.com> www.documentaccessibility.com

**Comment by nico3333fr** [comment]
Last commit has fixed the issue on VO.

Thanks for pointing out this bug. 👍 

For the "close it", it is more an overlay for users using the mouse, not really a link or button. Keyboard users should be trapped into the modal, when it is opened (focus is trapped even if the users tabulates from the URL bar).
